test/test_tensor_creation_ops.py

```diff
@@ -1960,6 +1960,54 @@ def test_inference(default_dtype):
         test_inference(torch.float64)
         test_inference(torch.float32)
 
+    @suppress_warnings
+    @onlyCPU
+    def test_tensor_factory_ndarrays(self, device):
+        """
+        This tests the case when the input to torch.tensor are multiple
+        numpy arrays that are embedded (recursively) within list(s). 
+        """
+
+        repeats = 10
+        np_types = [np.int8, np.int16, np.int32, np.int64, np.float16, np.float32,\
+                    np.float64, np.complex64, np.complex128]
+
+        for dtype in np_types: # Simple test
+            for _ in range(repeats):
+                arrays = [(10*np.random.randn(100)).astype(dtype) for _ in range(repeats)]
+                self.assertEqual(torch.tensor(arrays, device=device), torch.tensor(np.stack(arrays), device=device), exact_dtype=True, rtol=1e-05, atol=1e-08)
```

Comment:
Are there more distinct code paths to validate the correctness of torch.tensor() when given ndarrays? Maybe a Python reference implementation that converts each NumPy array to a tensor first? Or using torch.from_numpy()?

Reply:
Its a little cumbersome. A different path than using numpy.stack would have been to use torch's equivalent function but its stack function does not support recursively adding numpy arrays within lists. I changed ```torch.tensor``` to ```torch.from_numpy```